### PR TITLE
feat(wallet): add group-by-collection to portfolio NFTs page

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/38865

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Create a wallet with multiple NFTs for a collection
2. Open wallet portfolio and click "NFTs"
3. click the filters icon
4. observe the available settings
5. Enable "Group by collection" toggle
6. The NFTs portfolio page should update to display NFTs grouped by collection (may take a few moments to fetch token metadata needed to display the correct collection name)
7. Clicking a collection should navigate to a page that displays all NFTs in that collection

- Search should filter the NFT and NFT collections list
- Saving portfolio filters/settings (accounts, networks) should reset the portfolio NFTs page page-number to "1"
- Reloading the page should initially display the cached NFT collection names until new data is available
- The "Group by collection" toggle should not appear when filtering from the "Assets" tab (only on the NFTs tab)
-

### CODE REVIEW

**Review by yrliou — APPROVED** [review]
core++

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx [hunk 1]
```diff
@@ -89,6 +104,10 @@ import {
   ControlBarWrapper,
   ContentWrapper
 } from '../../portfolio/style'
+import {
+  getLastPageNumber,
+  getListPageItems
+} from '../../../../../utils/pagination_utils'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx:110]: Can move up to `// utils`

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-collection-grid-view-item.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// Styled Components
+import { NFTWrapper, NFTText, DIVForClickableArea } from './style'
+import { Row } from '../../../../../shared/style'
+import { NftIcon } from '../../../../../shared/nft-icon/nft-icon'
+import {
+  CollectionGrid,
+  EmptyCollectionGridItem,
+  NftIconStyles
+} from './nft-collection-grid-view-item.styles'
+import { getAssetIdKey } from '../../../../../../utils/asset-utils'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-collection-grid-view-item.tsx:20]: Can move to `// Utils`

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx (hunk 1 above)
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx]: moved

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-collection-grid-view-item.tsx (hunk 2 above)
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-collection-grid-view-item.tsx]: moved

**Review by Douglashdaniel — APPROVED** [review]
++

**Pull request merged**